### PR TITLE
#21641  removing all  gson references

### LINKED ISSUE #21641: Remove GSON dependency
GSON dependency must be removed and Jackson should be used instead.

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
I'm Removing Gson dependency and all the TypeAdapters used  to generate json
The Objects used before to feed MarshalUtils needed to be annotated so they can be used in the same way but through Jackson

### CODE REVIEW

**Comment by erickgonzalez** [comment]
tests failing:
https://raw.githack.com/dotCMS/test-results/issue-21641-remove-Gson/projects/core/current/unit/reports/html/index.html
https://raw.githack.com/dotCMS/test-results/issue-21641-remove-Gson/projects/core/current/integration/postgres/reports/html/classes/com.dotmarketing.portlets.templates.business.TemplateAPITest.html#findPagesByTemplate (This one I'm not sure if is b/c of the changes but I haven't seen this fail in master).

**Comment by fabrizzio-dotCMS** [comment]
> tests failing: https://raw.githack.com/dotCMS/test-results/issue-21641-remove-Gson/projects/core/current/unit/reports/html/index.html https://raw.githack.com/dotCMS/test-results/issue-21641-remove-Gson/projects/core/current/integration/postgres/reports/html/classes/com.dotmarketing.portlets.templates.business.TemplateAPITest.html#findPagesByTemplate (This one I'm not sure if is b/c of the changes but I haven't seen this fail in master).

didn't fail in previous runs
I'm gonna kick it again

**Review by nollymar — APPROVED** [review]

**Comment by erickgonzalez** [comment]
Unit tests still failing https://raw.githack.com/dotCMS/test-results/issue-21641-remove-Gson/projects/core/current/unit/reports/html/index.html

**Comment by fabrizzio-dotCMS** [comment]
> 

@erickgonzalez Yeah I had renamed that class and then I decided to put it the way it was.. and I lost the changes on that Unit Test.. It should be good now

File: dotCMS/src/integration-test/java/com/dotcms/content/business/ObjectMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Here a small test created for the purpose of experimenting wight jackson mapper and it's capabilities
+ */
+public class ObjectMapperTest {
```
Comment (erickgonzalez) [inline dotCMS/src/integration-test/java/com/dotcms/content/business/ObjectMapperTest.java:16]: This new test needs to be added to the MainSuite in order to run.

**Review by erickgonzalez — CHANGES_REQUESTED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/content/business/ObjectMapperTest.java (hunk 1 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/integration-test/java/com/dotcms/content/business/ObjectMapperTest.java:16]: @erickgonzalez done. Please continue with your review

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImpl.java [hunk 2]
```diff
@@ -761,13 +763,14 @@ private void addBulkRequest(final BulkIndexWrapper bulk, final List<Contentlet>
...
                     .sneak(() -> APILocator.getIndiciesAPI().loadIndicies());
-            final Gson gson = new Gson(); // todo why do we create a new Gson everytime
             String mapping = null;
 
             try {
 
                 if (this.isWorking(contentlet)) {
-                    mapping = gson.toJson(mappingAPI.toMap(contentlet));
+
+                    mapping = Try.of(()->objectMapper.writeValueAsString(mappingAPI.toMap(contentlet))).getOrElseThrow(
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImpl.java:772]: Probably should encapsulate it into a class (similar concept of the marshall utils)

**Review by jdotcms — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**